crates/tx5/tests/tests/flaky_sig.rs

```diff
@@ -0,0 +1,287 @@
+use crate::tests::{enable_tracing, receive_next_message_from, sbd};
+use futures::executor::block_on;
+use std::sync::Arc;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::sync::Mutex;
+use tokio::task::AbortHandle;
+use tx5::{Endpoint, EndpointRecv, PeerUrl};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn reconnect_on_all_signal_ws_disconnect() {
+    enable_tracing();
+
+    let sig = sbd().await;
+
+    let flaky_relay = FlakyRelay::connect(sig.bind_addrs()[0].clone()).await;
+
+    let (p1, e1, mut r1) = ep_for_relay(flaky_relay.bound_addr.clone()).await;
+    let (p2, e2, mut r2) = ep_for_relay(flaky_relay.bound_addr.clone()).await;
+
+    e1.send(p2.clone(), b"hello".to_vec()).await.unwrap();
+    let msg = receive_next_message_from(&mut r2, p1.clone()).await;
+    assert_eq!("hello", String::from_utf8_lossy(&msg));
+
+    e2.send(p1.clone(), b"world".to_vec()).await.unwrap();
+    let msg = receive_next_message_from(&mut r1, p2.clone()).await;
+    assert_eq!("world", String::from_utf8_lossy(&msg));
+
+    // Drop the connections from our endpoints to the signal server via the relay.
+    flaky_relay.just_lose_it().await;
+
+    // Now close the direct connections because those would survive the loss of the relay.
+    e1.close(&p2);
+    e2.close(&p1);
+
+    wait_for_connected(&e1, &e2, &mut r1, &mut r2).await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn reconnect_on_one_side_signal_ws_disconnect() {
```

Comment:
One side loses connectivity with the signal server and one side closes their p2p connection. Then do the same checks that the peers end up reconnected and able to send messages to each other.